tasks/main.yml

```diff
@@ -8,6 +8,7 @@
 - name: Download get-pip.py
   get_url: url=https://bootstrap.pypa.io/get-pip.py
            dest=/tmp/get-pip.py
+  when: pip_version_output | failed or not pip_version_output.stdout | search(pip_version)
```

Comment:
It may be worth looking into [blocks](https://www.pandastrike.com/posts/20160308-ansible-blocks-examples) as a way to avoid DRY patterns. With the chance of effecting readability, and how small the role is, this is really insignificant.